selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/NewWorkspace.java

```diff
@@ -185,7 +186,12 @@ public boolean isDevfileVisible(Devfile devfile) {
 
   public void selectDevfile(Devfile devfile) {
     waitDevfiles(asList(devfile));
-    seleniumWebDriverHelper.waitAndClick(By.xpath(format(DEVFILE_ROW_XPATH, devfile.getId())));
+    try {
+      seleniumWebDriverHelper.waitAndClick(By.xpath(format(DEVFILE_ROW_XPATH, devfile.getId())));
+      waitDevfileSelected(devfile);
+    } catch (WebDriverException ex) {
```

Comment:
Can we be more specific here, not to cover unrecoverable exceptions, but timeout exception?